CMake build breaks with gstreamer found because of Microhard/Taisync

## Expected Behavior
Build files can be built.

## Current Behavior
Link phase fails with
```
LINK : fatal error LNK1104: cannot open file 'Microhard.lib'
```

## Steps to Reproduce:
Please provide an unambiguous set of steps to reproduce the current behavior
1. Generate build files with gstreamer enabled
2. Leave Microhard and Taisync disabled

## System Information
When posting bug reports, include the following information
- Windows 10
- current master (54b15e48c6f569ea955b03d020c2e72180b53322)
- CMake build

## Detailed Description
The problem comes with changes made in PR #9733 and #9736.
An option was added to enable/disable Microhard and Taisync when gstreamer is found.
If gstreamer is found and Microhard or Taisync is off, they are still added in `src/CMakeLists.txt` line 195.

```
if(GST_FOUND)
	target_link_libraries(qgc PUBLIC Microhard)
	target_link_libraries(qgc PUBLIC Taisync)
endif()
```
When build files are generated, the linker now tries to find those libraries. But they are excluded in other places, because the options for Microhard and Taisync are disabled.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,6 +111,10 @@ if (GST_FOUND)
 
     option(QGC_GST_MICROHARD_ENABLED "Enable microhard" OFF)
     option(QGC_GST_TAISYNC_ENABLED "Enable taisyng" OFF)
+else()
+	if (QGC_GST_MICROHARD_ENABLED OR QGC_GST_TAISYNC_ENABLED)
+		message(FATAL_ERROR "You tried to enable Microhard or Taisync but gstreamer is not found. Make sure to set PKG_CONFIG_EXECUTABLE and/or PKG_CONFIG_PATH properly.")
+	endif()
 endif()
 
 add_definitions(

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -193,8 +193,14 @@ target_link_libraries(qgc
 )
 
 if(GST_FOUND)
-	target_link_libraries(qgc PUBLIC Microhard)
-	target_link_libraries(qgc PUBLIC Taisync)
+	if (${QGC_GST_MICROHARD_ENABLED})
+		target_compile_definitions(qgc PUBLIC QGC_GST_MICROHARD_ENABLED)
+		target_link_libraries(qgc PUBLIC Microhard)
+	endif()
+	if (${QGC_GST_TAISYNC_ENABLED})
+		target_compile_definitions(qgc PUBLIC QGC_GST_TAISYNC_ENABLED)
+		target_link_libraries(qgc PUBLIC Taisync)
+	endif()
 endif()
 
 if(BUILD_TESTING)

